libraries/rush-lib/src/cli/actions/BaseInstallAction.ts

```diff
@@ -143,18 +143,23 @@ export abstract class BaseInstallAction extends BaseRushAction {
 
     // If we are doing a filtered install and subspaces is enabled, we need to find the affected subspaces and install for all of them.
     let selectedSubspaces: ReadonlySet<Subspace> | undefined;
+    const filterArgumentsForSubspace: Map<Subspace, string[]> = new Map();
     if (this.rushConfiguration.subspacesFeatureEnabled) {
       if (installManagerOptions.pnpmFilterArguments.length) {
         // Selecting a set of subspaces
         const selectedProjects: Set<RushConfigurationProject> | undefined =
           await this._selectionParameters?.getSelectedProjectsAsync(this._terminal);
         if (selectedProjects) {
+          // Go through each project, add it to it's subspace's pnpm filter arguments
+          for (const project of selectedProjects) {
+            const subspaceFilterArguments: string[] = filterArgumentsForSubspace.get(project.subspace) || [];
+            subspaceFilterArguments.push('--filter', project.packageName);
+            filterArgumentsForSubspace.set(project.subspace, subspaceFilterArguments);
```

Comment:
Will this initialize arguments for non-selected subspaces? Should that be an error case?

Reply:
This runs when the user already provides filter arguments, for example `rush install --to <project_a>`, currently we are selecting the set of subspaces impacted by this, such as all the dependencies of project_a in other subspaces. So in this case, the set of `selectedProjects` should only include projects that are selected by the user's original filter command.